runtime/zsr/introspectable-json.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <zsr/introspectable.hpp>
+#include <zsr/types.hpp>
+
+#include <nlohmann/json.hpp>
+
+namespace zsr
+{
+
+enum ZSR_EXPORT SerializationOptions
+{
+    SERIALIZE_DEFAULT =           0u, /// Default serialization without metadata
+    SERIALIZE_TYPE =         1u << 0, /// Serialize TypeRef to '__type'
+    SERIALIZE_METADATA =     1u << 1, /// Serialize Compound to '__meta'
+    SERIALIZE_FUNCTIONS =    1u << 2, /// Serialize function return values
+    SERIALIZE_RESOLVE_ENUM = 1u << 3, /// Resolve enum values to key ident
+    SERIALIZE_BITBUFFER =    1u << 4, /// Serialize bitbuffers as arrays
+};
+
+/**
+ * Serialize an zsr object to JSON.
+ *
+ * \param obj   Object to serialize.
+ * \param pkgs  Optional package list, needed for type-lookup.
+ *              Needed for options: *_RESOLVE_ENUM.
+ * \param opt   Serialization options.
+ * \return      JSON object
+ */
+nlohmann::json ZSR_EXPORT serialize(const zsr::Introspectable& obj,
```

Comment:
Please add an example to the docstring of what a resulting json looks like with the different options enabled 😸